forte/data/ontology/top.py

```diff
@@ -1168,12 +1172,125 @@ def __init__(
         )
 
 
+class Payload(Entry):
+    """
+    A payload class that holds data cache of one modality and its data source uri.
+
+    Args:
+        pack: The container that this `Payload` will
+            be added to.
+        modality: modality of the payload such as text, audio and image.
+        payload_idx: the index of the payload.
+        uri: universal resource identifier of the data source. Defaults to None.
+
+    Raises:
+        ValueError: raised when the modality is not supported.
+    """
+
+    def __init__(
+        self,
+        pack: PackType,
+        modality: IntEnum,
+        payload_idx: int = 0,
+        uri: Optional[str] = None,
+    ):
+        supported_modality = ("text", "audio", "image")
+        if modality.name not in supported_modality:
+            raise ValueError(
+                f"The given modality {modality} is not supported. "
+                f"Currently we only support {supported_modality}"
+            )
+        self._payload_idx: int = payload_idx
+        self._modality: IntEnum = modality
+        self._uri: Optional[str] = uri
+
+        super().__init__(pack)
+        # self._cache: Optional[Union[str, np.ndarray]] = None
+        self._cache: Union[str, np.ndarray] = ""
```

Comment:
default to empty string here is weird, probably `None`

Reply:
For backward compatibility, there are many cases `TextPayload.cache` is expected to be an empty string.